core/server/api/index.js

```diff
@@ -40,6 +40,14 @@ var _ = require('lodash'),
     contentDispositionHeaderRedirects,
     contentDispositionHeaderRoutes;
 
+function containsId(object) {
+    return Object.keys(object).some(function (key) {
+        return object[key].some(function hasId(obj) {
```

Comment:
As far as I can see we only use `data[resource][0]` for writes, so this could be optimised